src/pages/home/ReportScreen.js

```diff
@@ -293,7 +291,7 @@ class ReportScreen extends React.Component {
                                     ) : (
                                         <HeaderView
                                             reportID={reportID}
-                                            onNavigationMenuButtonClicked={() => Navigation.navigate(ROUTES.HOME)}
+                                            onNavigationMenuButtonClicked={Navigation.goBack}
```

Comment:
This caused a regression https://github.com/Expensify/App/issues/20370. After viewing many reports (that are not threaded nor linked to each other i.e. any random reports from LHN) then we click the `<` button on the last report, we will navigate back to the previous report instead of LHN. This is kinda an edge case I think.